scylla-cql/src/frame/frame_errors.rs

```diff
@@ -108,6 +108,24 @@ pub enum PreparedParseError {
     PreparedMetadataParseError(ResultMetadataParseError),
 }
 
+/// An error type returned when deserialization
+/// of `RESULT::Rows` response fails.
+#[non_exhaustive]
+#[derive(Debug, Error)]
+pub enum RowsParseError {
+    #[error("Invalid result metadata: {0}")]
+    ResultMetadataParseError(#[from] ResultMetadataParseError),
+    #[error("Invalid result metadata, expected {expected_col_count} col specs, received {received_col_count}.")]
+    ColumnCountMismatch {
+        expected_col_count: usize,
+        received_col_count: usize,
+    },
+    #[error("Malformed rows count: {0}")]
+    RowsCountParseError(LowLevelDeserializationError),
+    #[error("Data deserialization failed: {0}")]
+    DataDeserializationError(#[from] DeserializationError),
```

Comment:
Is it specific enough? Does the upper layer add information that it's about data received in RESULT::ROWS response?